docs/src/rules/no-unassigned-vars.md

```diff
@@ -0,0 +1,107 @@
+---
+title: no-unassigned-vars
+rule_type: problem
+related_rules:
+- prefer-const
+- init-declarations
+---
+
+
+This rule flags `let` or `var` declarations that have no initializer and are never assigned any value, but are read/used in the code. These variables will always be `undefined` and are likely a programmer error.
```

Comment:
```suggestion
This rule flags `let` and `var` declarations that have no initializer and are read without first being assigned a value. These variables will always be `undefined` and are likely a programmer error.
```

Reply:
We should also show an example here. This intro is supposed to describe what the rule does with an example.

Reply:
The suggested wording would be inaccurate, since the rule is not checking the order of reads and writes, it is specifically checking for the case where there are some reads but there are no writes whatsoever (and no initializer).